### PR TITLE
Set --service-account-issuer for k8s 1.20+

### LINKED ISSUE #10279: Clusters created with k8s v1.20 don't start
e2e tests for k8s v1.20 fail to start:
https://testgrid.k8s.io/kops-versions#kops-aws-k8s-latest

Me and @rifelpet tracked this to https://github.com/kubernetes/kubernetes/pull/95896, which introduces some new mandatory flags:
```
TokenRequest and TokenRequestProjection are now GA features. The following flags are required by the API server:
- `--service-account-issuer`, should be set to a URL identifying the API server that will be stable over the cluster lifetime.
- `--service-account-key-file`, set to one or more files containing one or more public keys used to verify tokens.
- `--service-account-signing-key-file`, set to a file containing a private key to use to sign service account tokens. Can be the same file given to `kube-controller-manager` with `--service-account-private-key-file`.
```

/cc @justinsb @johngmyers @olemarkus

### PR DESCRIPTION
**Pull request by johngmyers**
Kubernetes 1.20+ requires the `--service-account-issuer` flag be passed to kube-apiserver

Fixes #10279 

/cc @justinsb

### CODE REVIEW

File: pkg/model/iam/subject.go [hunk 1]
```diff
@@ -83,11 +82,7 @@ func BuildNodeRoleSubject(igRole kops.InstanceGroupRole) (Subject, error) {
 
 // ServiceAccountIssuer determines the issuer in the ServiceAccount JWTs
 func ServiceAccountIssuer(clusterName string, clusterSpec *kops.ClusterSpec) (string, error) {
-	if featureflag.PublicJWKS.Enabled() {
-		return "https://api." + clusterName, nil
-	}
-
-	return "", fmt.Errorf("ServiceAcccountIssuer not (currently) supported without PublicJWKS")
+	return "https://api." + clusterName, nil
```
Comment (johngmyers) [inline pkg/model/iam/subject.go:85]: Somewhat unrelated to what this PR is trying to fix, if the admin has set `spec.kubeAPIServer.serviceAccountIssuer` explicitly, this should probably return that value instead.

**Comment by johngmyers** [comment]
This is at least sufficient to get k8s 1.20 to start, but I'm not entirely confident about setting that issuer without enabling discovery. I would appreciate review from @justinsb or @rifelpet.
/approve cancel

**Comment by olemarkus** [comment]
Service account issuer discovery was promoted to beta in 1.20 so this feature should be enabled by default. I don't think we need to do anything in particular around that feature.

File: pkg/model/iam/subject.go (hunk 1 above)
Comment (rifelpet) [reply pkg/model/iam/subject.go:85]: I agree we should respect the API field but that could be done in a separate PR.

This function no longer needs to return an error

**Review by rifelpet — COMMENTED** [review]
this looks good to me but I'd like Justin's opinion on potential impact

/cc @justinsb

**Comment by justinsb** [comment]
This looks good.  One thing we'll have to verify is that upgrading works!

/approve
/lgtm

**Pull request merged**